File: src/main/java/no/nav/bidrag/beregn/saertilskudd/SaertilskuddCore.java
package no.nav.bidrag.beregn.saertilskudd;

import no.nav.bidrag.beregn.saertilskudd.dto.BeregnSaertilskuddGrunnlagCore;
import no.nav.bidrag.beregn.saertilskudd.dto.BeregnSaertilskuddResultatCore;
import no.nav.bidrag.beregn.saertilskudd.periode.SaertilskuddPeriode;


public interface SaertilskuddCore {

  BeregnSaertilskuddResultatCore beregnSaertilskudd(BeregnSaertilskuddGrunnlagCore beregnSaertilskuddGrunnlagCore);

  static SaertilskuddCore getInstance() {
    return new SaertilskuddCoreImpl(SaertilskuddPeriode.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/saertilskudd/SaertilskuddCore.kt
package no.nav.bidrag.beregn.saertilskudd

import no.nav.bidrag.beregn.saertilskudd.dto.BeregnSaertilskuddGrunnlagCore
import no.nav.bidrag.beregn.saertilskudd.dto.BeregnSaertilskuddResultatCore
import no.nav.bidrag.beregn.saertilskudd.periode.SaertilskuddPeriode

fun interface SaertilskuddCore {
    fun beregnSaertilskudd(grunnlag: BeregnSaertilskuddGrunnlagCore): BeregnSaertilskuddResultatCore

    companion object {
        fun getInstance(): SaertilskuddCore = SaertilskuddCoreImpl(SaertilskuddPeriode.getInstance())
    }
}
